client/js/table-view.js

```diff
@@ -2,146 +2,165 @@ const { getLetterRange } = require('./array-util');
 
-    renderTable() {
-        this.renderTableHeader();
-        this.renderTableBody();
-        this.renderTableFooter();
+    const tr = createTR();
+    for (let col = 0; col < this.model.numCols; col++) {
+      const td = createTD('');
+      tr.appendChild(td);
     }
 
-    renderTableHeader() {
-        // clear header row
-        removeChildren(this.headerRowEl);
-        // get letters and build elements
-        getLetterRange('A', this.model.numCols)
-            .map(colLabel => createTH(colLabel))
-            .forEach(th => this.headerRowEl.appendChild(th));
-    }
+    fragment.appendChild(tr);
 
+    this.sheetBodyEl.appendChild(fragment);
+  }
 
-    isCurrentCell(col, row) {
-        return this.currentCellLocation.row === row &&
-            this.currentCellLocation.col === col;
-    }
 
-    renderTableBody() {
-        const fragment = document.createDocumentFragment();
-        for (let row = 0; row < this.model.numRows; row++) {
-            const tr = createTR();
-            for (let col = 0; col < this.model.numCols; col++) {
-                const position = { col: col, row: row };
-                const value = this.model.getValue(position);
-                const td = createTD(value);
-
-                if (this.isCurrentCell(col, row)) {
-                    td.className = 'current-cell';
-                }
-
-                tr.appendChild(td);
-            }
-            fragment.appendChild(tr);
-        }
-        removeChildren(this.sheetBodyEl);
-        this.sheetBodyEl.appendChild(fragment);
-    }
 
-    renderTableFooter() {
-        // clear header row
-        removeChildren(this.footerRowEl);
-        // get letters and build elements
-        for (let col = 0; col < this.model.numCols; col++) {
-            const position = { col: col, row: this.model.numRows };
-            const value = this.model.getValue(position);
-            const td = createTD(value);
-            this.footerRowEl.appendChild(td);
-        }
-    }
+  handleFormulaBarChange(evt) {
+    const value = this.formulaBarEl.value;
+    this.model.setValue(this.currentCellLocation, value);
+    this.model.setValue(this.getFooterCellLocation(), this.sumColumn());
+    this.renderTableBody();
+    this.renderTableFooter();
+  }
 
-    attachEventHandlers() {
-        this.sheetBodyEl.addEventListener('click', this.handleSheetClick.bind(this));
-        this.formulaBarEl.addEventListener('keyup', this.handleFormulaBarChange.bind(this));
-        this.addColumnButton.addEventListener('click', this.handleAddColumn.bind(this));
-        this.addRowButton.addEventListener('click', this.handleAddRow.bind(this));
-    }
+  getFooterCellLocation() {
+    return { col: this.currentCellLocation.col, row: this.model.numRows };
+  }
 
-    handleAddColumn() {
-        this.model.numCols++;
-        this.renderTable();
-    }
+  sumColumn() {
+    let sum = 0;
+    for (let i = 0; i < this.model.numRows; i++) {
```

Comment:
You have used "let row = 0" in other places, and that would be better here as well. More consistency and more readability === better code.